client/src/components/Project/Scrums/index.jsx

```diff
@@ -0,0 +1,264 @@
+import React, { Component } from 'react';
+import { Table, Button, Icon } from 'antd';
+import Swal from 'sweetalert2';
+import Editable from 'react-contenteditable';
+import {StatusSelect, ActionTypeSelect} from './select/index';
+import { Sort } from './utilis/sort';
+import { Filter } from './utilis/filter.js'
+const tasks = require('./utilis/tasks');
+
+class Scrum extends Component {
+    state = {
+        tasks: [],
+        html: '',
+        newRow: {},
+        saving: false,
+        saved: false,
+        taskDescriptionErr: false,
+        rowSelected: null,
+    }
+
+    componentDidMount() {
+        this.setState({tasks: tasks})
+    }
+
+    handleAddNewTask = () => {
+      return (
+          this.setState({
+          tasks: [...tasks, { 
+              id: tasks.length + 1, 
+              task_name: '', 
+              task_description: '', 
+              action_type: 'testing' ,
+              priority:'',
+              est_time:'',
+              remaining_time:'',
+              status:'in progress',
+              assignee:'',
+              ticket:''
+          }],
+        })
+      )
+    };
+
+    handleAddTask = (event, column) => {
+      const newTask = event.target.value;
+      this.setState(prevState => {
+        const clonedTasks = [...tasks];
+        clonedTasks[clonedTasks.length - 1][column] = newTask;
+        return { tasks: clonedTasks, newRow: clonedTasks[clonedTasks.length - 1] };
+      });
+    };
+
+    handleEditTask = (event, record, column) => {
+      const { tasks } = this.state;
+
+      //Add a new task
+      if (record.id === tasks[tasks.length - 1].id) {
+        this.setState({ saving: true, saved: false })
+        return this.handleAddTask(event, column);
+      }
+
+      //Edit Task
+      const newTask = event.target.value;
+      const taskId = record.id;
+      const clonedTasks = [...tasks];
+      const updateTask = clonedTasks.find(task => task.id === taskId);
+      updateTask[column] = newTask;
+
+      //Validate
+      if (this.validateTask(updateTask)) {
+        this.updateTasks(updateTask, clonedTasks);
+      }
+    };
+
+    updateTasks = (task, tasks) => {
+      this.setState({ tasks, saved: true });
+    };
+
+    validateTask = (task) => {
+      this.setState({ taskDescriptionErr: false });
+      if (task['task_description'].length < 1) {
+        this.setState({ taskDescriptionErr: true });
+        return false;
+      }
```

Comment:
[check this](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous)

**Also,** this function will always return a falsey value. either by `return false;` or by not returning anything